### PR TITLE
feat(terraform): Adding yaml based build time policies for corresponding PC runtime policies

### PR DESCRIPTION
**Pull request by praveen-panw**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

This PR has 5 YAML-based checks with detailed description below

### 1. Checkov Title - Ensure Azure SQL server firewall is not overly permissive
 
<ins>PC Policy ID</ins> - 0faffeb6-dbcd-4715-be6e-f9cadc64cfeb
 
<ins>PC Policy Title</ins> - Azure SQL Server allow access to any Azure internal resources
 
<ins>Compliance standards </ins> - 
 
`HITRUST CSF v9.3` `PCI DSS v3.2.1` `CCPA 2018` `CyberSecurity Law of the People's Republic of China` `Copy of CCPA 2018-0417` `ISO 27001:2013` `SOC 2` `CSA CCM v3.0.1` `CIS v1.2.0 (Azure)` `CIS v1.3.0 (Azure)` `Brazilian Data Protection Law (LGPD)` `MITRE ATT&CK v8.2` `NIST SP 800-171 Revision 2` `NIST SP 800-172` `NIST CSF` `CSA CCM v.4.0.1` `CIS v1.5.0 (Azure) - Level 1` `Fedramp (Moderate)` `NIST CSF (中文)` `Azure Security Benchmark (v2)` `ISO/IEC 27002:2013` `Fedramp (Low)` `New Zealand Information Security Manual (NZISM v3.4)` `ISO/IEC 27018:2019` `Azure Security Benchmark (v3)` `Copy of (AESCSF)-IR` `CIS Controls v7.1` `FFIEC` `PCI DSS v4.0` `Australian Energy Sector Cyber Security Framework (AESCSF)` `ACSC Information Security Manual (ISM)` `Copy of ACSC Information Security Manual (ISM)` `MLPS 2.0 (Level 2)` `Copy 1 0 of ACSC Information Security Manual (ISM)` `Copy 11 of ACSC Information Security Manual (ISM)` `ISO 27002:2022` `ISO/IEC 27001:2022` `CIS v2.0.0 (Azure) Level 1` `MITRE ATT&CK v12` `PIPEDA` `NIST 800-53 Rev4` `CIS v1.1 (Azure)` `MITRE ATT&CK v6.3` `APRA (CPS 234) Information Security` `Cybersecurity Maturity Model Certification (CMMC) v.1.02` `CIS v1.3.1 (Azure)` `Sarbanes Oxley Act (SOX)` `ISO/IEC 27017:2015` `MITRE ATT&CK v10.0` `CIS v1.4.0 (Azure)` `CIS Controls v8` `Secure Controls Framework (SCF) - 2022.2.1`
 
<ins>Remediation:</ins>
 
1. Log in to Azure Portal
2. Navigate to the 'SQL servers' dashboard
3. Click on the reported SQL server
4. Click on 'Networking' under Security
5. Unselect 'Allow Azure services and resources to access this server' under Exceptions if selected.
6. Remove any firewall rule which allows access to 0.0.0.0 in startIpAddress and endIpAddress if any.
7. Click on 'Save'
### 2. Checkov Title - Ensure Azure recovery services vault is configured with managed identity
 
<ins>PC Policy ID</ins> - aa72a7b0-65f3-4e15-8b62-fb3b6886f67a
 
<ins>PC Policy Title</ins> - Azure Recovery Services vault is not configured with managed identity
 
<ins>Compliance standards </ins> - 
 
`CIS Controls v7.1` `PCI DSS v4.0` `Australian Energy Sector Cyber Security Framework (AESCSF)` `ACSC Information Security Manual (ISM)` `Copy of ACSC Information Security Manual (ISM)` `MLPS 2.0 (Level 2)` `Copy 11 of ACSC Information Security Manual (ISM)` `Sarbanes Oxley Act (SOX)` `ISO 27002:2022` `ISO/IEC 27001:2022` `New Zealand Information Security Manual (NZISM v3.4)` `CIS Controls v8` `Copy 1 0 of ACSC Information Security Manual (ISM)` `Secure Controls Framework (SCF) - 2022.2.1`
 
<ins>Remediation:</ins>
 
1. Log in to the Azure portal
2. Navigate to Recovery Services vaults dashboard
3. Click on the reported Recovery Services vault
4. Under Setting section, Click on 'Identity'
5. Configure either 'System assigned' or 'User assigned' managed identity based on your requirement.
6. Click on 'Save'
### 3. Checkov Title - Ensure Azure automation account is configured with managed identity
 
<ins>PC Policy ID</ins> - 76edf731-a606-4409-86a9-6f71209b6681
 
<ins>PC Policy Title</ins> - Azure Automation account is not configured with managed identity
 
<ins>Compliance standards </ins> - 
 
`MLPS 2.0 (Level 2)` `Sarbanes Oxley Act (SOX)` `Secure Controls Framework (SCF) - 2022.2.1`
 
<ins>Remediation:</ins>
 
To enable managed identity on an existing Azure Automation account, follow the below URL:
https://docs.microsoft.com/en-us/azure/automation/quickstarts/enable-managed-identity
### 4. Checkov Title - Ensure Azure MariaDB server is using latest TLS (1.2)
 
<ins>PC Policy ID</ins> - ba4d0c4a-e02b-4f89-a916-f5fdfeb83f7a
 
<ins>PC Policy Title</ins> - Azure MariaDB database server not using latest TLS version
 
<ins>Compliance standards </ins> - 
 
`New Zealand Information Security Manual (NZISM v3.4)` `CIS Controls v8` `FFIEC` `Copy of ACSC Information Security Manual (ISM)` `Secure Controls Framework (SCF) - 2022.2.1` `Sarbanes Oxley Act (SOX)` `CSA CCM v.4.0.6` `ISO 27002:2022` `ISO/IEC 27001:2022` `CIS Controls v7.1` `PCI DSS v4.0` `ACSC Information Security Manual (ISM)` `MLPS 2.0 (Level 2)` `Copy 1 0 of ACSC Information Security Manual (ISM)` `Copy 11 of ACSC Information Security Manual (ISM)`
 
<ins>Remediation:</ins>
 
To configure SSL connection with latest TLS version on an existing Azure Database for MariaDB, follow the below URL:
https://docs.microsoft.com/en-us/azure/mariadb/howto-tls-configurations

NOTE: Once you enforce a minimum TLS version, you cannot later disable minimum version enforcement.
### 5. Checkov Title - Ensure soft-delete is enabled on Azure storage account
 
<ins>PC Policy ID</ins> - f5a29936-659e-48a8-8110-783411bf6a9c
 
<ins>PC Policy Title</ins> - Azure Storage account soft delete is disabled
 
<ins>Compliance standards </ins> - 
 
`CIS v1.2.0 (Azure)` `CIS v1.3.0 (Azure)` `NIST SP 800-171 Revision 2` `NIST SP 800-172` `HITRUST v.9.4.2` `NIST CSF` `CSA CCM v.4.0.1` `ISO/IEC 27002:2013` `New Zealand Information Security Manual (NZISM v3.4)` `ISO/IEC 27017:2015` `ISO/IEC 27018:2019` `CIS v1.4.0 (Azure)` `Copy of (AESCSF)-IR` `CIS Controls v7.1` `CIS Controls v8` `ACSC Information Security Manual (ISM)` `Copy of ACSC Information Security Manual (ISM)` `MLPS 2.0 (Level 2)` `Copy 1 0 of ACSC Information Security Manual (ISM)` `Copy 11 of ACSC Information Security Manual (ISM)` `Secure Controls Framework (SCF) - 2022.2.1` `Sarbanes Oxley Act (SOX)` `ISO 27002:2022` `ISO/IEC 27001:2022` `CIS v1.3.1 (Azure)` `PCI DSS v3.2.1` `NIST CSF (中文)` `MITRE ATT&CK v10.0` `Azure Security Benchmark (v3)` `MITRE ATT&CK v12`
 
<ins>Remediation:</ins>
 
To enable Soft delete on your storage account, follow below URL:
https://learn.microsoft.com/en-gb/azure/storage/blobs/soft-delete-blob-enable?tabs=azure-portal#enable-blob-soft-delete

### CODE REVIEW

File: checkov/terraform/checks/graph_checks/azure/AzureRecoveryServicesvaultConfigManagedIdentity.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+metadata:
+  id: "CKV2_AZURE_35"
+  name: "Ensure Azure recovery services vault is configured with managed identity"
+  category: "GENERAL_SECURITY"
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/azure/AzureRecoveryServicesvaultConfigManagedIdentity.yaml:4]:
```suggestion
  category: "IAM"
```

File: tests/terraform/graph/checks/resources/AzureAutomationAccConfigManagedIdentity/expected.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+pass:
+  - "azurerm_recovery_services_vault.pass"
+
+fail:
+  - "azurerm_recovery_services_vault.fail_1"
+  - "azurerm_recovery_services_vault.fail_2"
```
Comment (gruebel) [inline tests/terraform/graph/checks/resources/AzureAutomationAccConfigManagedIdentity/expected.yaml:6]:
```suggestion
  - "azurerm_automation_account.pass"

fail:
  - "azurerm_automation_account.fail_1"
  - "azurerm_automation_account.fail_2"
```

File: tests/terraform/graph/checks/resources/AzureMariaDBserverUsingTLS_1_2/main.tf [hunk 3]
```diff
@@ -0,0 +1,49 @@
+
+resource "azurerm_mariadb_server" "pass_1" {
+  name                = "pud-mariadb-server"
+  location            = azurerm_resource_group.pud_rg.location
+  resource_group_name = azurerm_resource_group.pud_rg.name
+
+  administrator_login          = "dbadmin123"
+  administrator_login_password = "M@r!@D3"
```
[the remark this answers is not in the document]
Comment (praveen-panw) [reply tests/terraform/graph/checks/resources/AzureMariaDBserverUsingTLS_1_2/main.tf:8]: Noted

**Review by gruebel — APPROVED** [review]
🍺

**Review by bo156 — APPROVED** [review]

**Pull request merged**